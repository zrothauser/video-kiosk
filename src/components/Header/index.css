.b-header {
  position: relative;
  height: 2.6041666667vw;
  background: var(--color__background--darker);
  line-height: 1;
  z-index: 4;
}

.b-header__title {
  font-size: 0.8333333333vw;
  margin: 0;
  float: left;
}

.b-header__index {
  width: 25%;
  margin: 0;
  float: right;
  cursor: pointer;
}

.b-header__link {
  -webkit-appearance: none;
  display: block;
  padding: 1.0108854167vw 1.5951041667vw;
  background: transparent;
  border: 0;
  color: var(--color__link);
  text-transform: uppercase;
  font-weight: 700;
  letter-spacing: .160em;
  font-size: 0.8333333333vw;
  line-height: 1;
}
