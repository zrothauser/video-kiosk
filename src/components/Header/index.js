// Dependencies
import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Transition from 'react-transition-group/Transition';

// Transition
import * as transitions from '../transitions';

// Styles
import './index.css';

const {
  durations,
  mediumFadeStyles,
} = transitions;

const Header = props => (
  <Transition
    in={props.visible}
    timeout={durations.medium}
  >
    {state => (
      <header
        className="b-header"
        style={{
          ...mediumFadeStyles.default,
          ...mediumFadeStyles[state],
        }}
      >
        <h1 className="b-header__title">
          <Link
            to={props.homeURL}
            className="b-header__link"
            onClick={() => props.closeVideoIndex()}
          >
            Clyfford Still Museum Media Channel
          </Link>
        </h1>

        <span className="b-header__index">
          <button
            onClick={() => props.toggleVideoIndex()}
            className="b-header__link"
          >
            Video Index
          </button>
        </span>
      </header>
    )}
  </Transition>
);

Header.propTypes = {
  toggleVideoIndex: PropTypes.func.isRequired,
  closeVideoIndex: PropTypes.func.isRequired,
  visible: PropTypes.bool.isRequired,
  homeURL: PropTypes.string.isRequired,
};

export default Header;
