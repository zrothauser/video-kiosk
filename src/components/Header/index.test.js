import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Header from './index';

describe('components', () => {
  describe('Header', () => {
    const mockProps = {
      toggleVideoIndex: jest.fn(),
      closeVideoIndex: jest.fn(),
      visible: true,
      homeURL: 'https://video.clyffordstillmuseum.org',
    };

    it('renders without crashing', () => {
      shallow(<Header {...mockProps} />);
    });

    it('renders correctly', () => {
      const HeaderComponent = renderer.create(
        <Router>
          <Header {...mockProps} />
        </Router>,
      );

      expect(HeaderComponent).toMatchSnapshot();
    });
  });
});
