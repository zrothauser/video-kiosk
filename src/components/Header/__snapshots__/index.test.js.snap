// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Header renders correctly 1`] = `
<header
  className="b-header"
  style={
    Object {
      "opacity": 1,
      "transition": "opacity 400ms ease-in-out",
    }
  }
>
  <h1
    className="b-header__title"
  >
    <a
      className="b-header__link"
      href="/https://video.clyffordstillmuseum.org"
      onClick={[Function]}
    >
      Clyfford Still Museum Video Channel
    </a>
  </h1>
  <span
    className="b-header__index"
  >
    <button
      className="b-header__link"
      onClick={[Function]}
    >
      Index
    </button>
  </span>
</header>
`;
